localflavor/cu/forms.py

```diff
@@ -0,0 +1,188 @@
+
+from .choices import PROVINCE_CHOICES, PROVINCE_NORMALIZED, REGION_CHOICES, REGION_NORMALIZED
+from .validators import CUIdentityCardNumberValidator
+
+
+class CURegionField(Field):
+    """
+    A form field for a cuban region.
+    The input is validated against a dictionary which includes names and abbreviations.
+
+    It normalizes the input to the standard abbreviation for the given region.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a cuban region.'),
+    }
+
+    def clean(self, value):
+        super(CURegionField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
+        try:
+            value = value.strip().lower()
+        except AttributeError:
+            pass
+        else:
+            try:
+                return REGION_NORMALIZED[value.strip().lower()]
+            except KeyError:
+                pass
+        raise ValidationError(self.error_messages['invalid'])
+
+
+class CURegionSelect(Select):
+    """
+    A Select widget that uses a list of cuban regions as its choices.
+
+    .. versionadded:: 1.6
+    """
+
+    def __init__(self, attrs=None):
+        super(CURegionSelect, self).__init__(attrs, choices=REGION_CHOICES)
+
+
+class CUProvinceField(Field):
+    """
+    A form field for a cuban province.
+    The input is validated against a dictionary which includes names and abbreviations.
+
+    It normalizes the input to the standard abbreviation for the given province.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a cuban province.'),
+    }
+
+    def clean(self, value):
+        super(CUProvinceField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
+        try:
+            value = value.strip().lower()
+        except AttributeError:
+            pass
+        else:
+            try:
+                return PROVINCE_NORMALIZED[value.strip().lower()]
+            except KeyError:
+                pass
+        raise ValidationError(self.error_messages['invalid'])
+
+
+class CUProvinceSelect(Select):
+    """
+    A Select widget that uses a list of cuban provinces as its choices.
+
+    .. versionadded:: 1.6
+    """
+
+    def __init__(self, attrs=None):
+        super(CUProvinceSelect, self).__init__(attrs, choices=PROVINCE_CHOICES)
+
+
+class CUPostalCodeField(RegexField):
+    """
+    A form field for a cuban Zip Code.
+
+    Taken from : http://mapanet.eu/Postal_Codes/?C=CU
+
+    The cuban ZIP code is a combination of 5 digits non begin with 0.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid zip code in the format XXXXX.'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super(CUPostalCodeField, self).__init__(r'^[1-9]\d{4}$', *args, **kwargs)
+
+    def to_python(self, value):
+        value = super(CUPostalCodeField, self).to_python(value)
```

Comment:
You need to add `localflavor.compat.EmptyValueCompatMixin` to this class and add this python code:

```python
if value in self.empty_values:
    return value
```

This comes from a recent change that was made to properly support Django >= 1.11 (#298). You can search through the other localflavor modules to see examples of how to set this up if you're not sure.